tidy3d/components/geometry/triangulation.py

```diff
@@ -41,15 +42,21 @@ def update_convexity(vertices: List[Vertex], i: int) -> None:
         Vertices of the polygon.
     i : int
         Index of the vertex to be updated.
+
+    Returns
+    -------
+    int
+        -1 if vertex was collinear, +1 if it became collinear, 0 otherwise.
```

Comment:
This part is hard to understand. Maybe add more explanation here?
My understanding by reading the code below is: -1 for vertex was collinear, and no longer collinear after the update. But what about +1 and 0?

Reply:
This return value is just to quickly update the counter of collinear vertices, i.e., if the update decreased the number of collinear vertices (a vertex was collinear and no longer is), it returns -1; if the update increased the number of collinears (the vertex was not collinear and now it is), it returns +1; else, the update doesn't change the count and it returns 0.

Reply:
Maybe just add this (or a short version of it) to the docstring @lucas-flexcompute ?

Reply:
Done! I think it's more clear now with the note and comment.

Reply:
Thanks, could you squash and merge.